velox/functions/sparksql/DecimalArithmetic.cpp

```diff
@@ -539,7 +585,11 @@ void registerDecimalAddSubtract(const std::string& name) {
       "max({a_scale}, {b_scale})",
       fmt::arg("a_scale", S1::name()),
       fmt::arg("b_scale", S2::name()));
-  registerDecimalBinary<Func>(name, makeConstraints(rPrecision, rScale));
+  registerDecimalBinary<Func, true>(
+      name, makeConstraints(rPrecision, rScale, true));
+  registerDecimalBinary<Func, false>(
+      name + "_not_allow_precision_loss",
```

Comment:
The suffix '_not_allow_precision_loss' is used several times. Consider defining a variable for it and renaming as '_deny_precision_loss'.